lib/browser-auto-tracking.js

```diff
@@ -87,6 +87,122 @@ export default function initAutoTrackingCore(lib) {
+              (y = l.getElementsByTagName(r)[0]).parentNode.insertBefore(t, y), a[c]("start", i),
+              a[c].q.unshift(a[c].q.pop());
+        })("clarity", document, window, "script", {
+          projectId: microsoft_project_id,
+          upload: clarity_upload_endpoint,
+          expire: 365,
+          cookies: ["_uetmsclkid", "_uetvid"],
+          track: true,
+          lean: false,
+          content: true,
+          extract: [3, 1, ["4f509md79", "zjy56jaz", "4s4zm72hj"]],
+        });
+      `;
+
+      !(function(c,l,a,r,i,t,y){
+        c[a]=c[a]||function(){
+          (c[a].q=c[a].q||[]).push(arguments)};
+          t=l.createElement(r);
+          t.async=1;
+          t.text=clarity_loading_script;
+          y=l.getElementsByTagName(r)[0];
+          y.parentNode.insertBefore(t,y);
+      })(window, document, "clarity", "script", microsoft_project_id);
+
+      //todo (clarity): how should we actually get/set cookies? (this is taken from clarity)
+      function getCookie(key) {
+        let cookies = document.cookie.split(';');
+        if (cookies) {
+          for (let i = 0; i < cookies.length; i++) {
+            let pair = cookies[i].split('=');
+            if (pair.length > 1 && pair[0] && pair[0].trim() === key) {
+              return pair[1];
+            }
+          }
+        }
+        return null;
+      }
+
+      let rootDomain = null;
+      function setCookie(key, value, time){
+        let expiry = new Date();
+        expiry.setDate(expiry.getDate() + time);
+        let expires = expiry ? "expires=" + expiry.toUTCString() : "";
+        let cookie = `${key}=${value};${expires};path=/`;
+        try {
+          // Attempt to get the root domain only once and fall back to writing cookie on the current domain.
+          if (rootDomain === null) {
+            let hostname = location.hostname ? location.hostname.split('.') : [];
+            // Walk backwards on a domain and attempt to set a cookie, until successful
+            for (let i = hostname.length - 1; i >= 0; i--) {
+              rootDomain = `.${hostname[i]}${rootDomain ? rootDomain : ""}`;
+              // We do not wish to attempt writing a cookie on the absolute last part of the domain, e.g. .com or .net.
+              // So we start attempting after second-last part, e.g. .domain.com (PASS) or .co.uk (FAIL)
+              if (i < hostname.length - 1) {
+                // Write the cookie on the current computed top level domain
+                document.cookie = `${cookie};domain=${rootDomain}`;
+                if (getCookie(key) === value) { return; }
+              }
+            }
+            // Finally, if we were not successful and gone through all the options, play it safe and reset rootDomain to be empty
+            // This forces our code to fall back to always writing cookie to the current domain
+            rootDomain = "";
+          }
+        } catch { rootDomain = ""; }
```

Comment:
For legacy sake you might want to make this catch () { rootDomain = ""; }. Or make the syntax less funny and just do this.
```
catch () {
  rootDomain = "";
}
```
https://caniuse.com/?search=optional%20catch%20binding

Reply:
My hope is to blow this whole function away and replace it with a library, but noted.